convert2rhel/unit_tests/actions/actions_test.py

```diff
@@ -0,0 +1,793 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright(C) 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import os.path
+import re
+
+from collections import defaultdict
+
+import pytest
+import six
+
+
+six.add_move(six.MovedModule("mock", "mock", "unittest.mock"))
+from six.moves import mock
+
+from convert2rhel import actions
+from convert2rhel.actions import STATUS_CODE
+
+
+class _ActionForTesting(actions.Action):
+    """Fake Action class where we can set all of the attributes as we like."""
+
+    id = None
+
+    def __init__(self, **kwargs):
+        super(_ActionForTesting, self).__init__()
+        for attr_name, attr_value in kwargs.items():
+            setattr(self, attr_name, attr_value)
+
+    def run(self):
+        super(_ActionForTesting, self).run()
+        pass
```

Comment:
```suggestion
```

Reply:
Can't quite do this as a different linting tool complains that `run()` doesn't do anything in that case.  Maybe the other lint tool will be silent if we put in a docstring?

Reply:
Nope, a docstring doesn't do it.  I think we should mark this as a false positive in CodeQL (We want the`pass` statement to mark that we are intentionally not changing any behaviour here.

Reply:
that works for me.